src/worker.rs

```diff
@@ -734,8 +794,22 @@ impl ClientWorker {
                     let sender = ev.sender().to_owned();
                     let _ = locked.application.presences.get_or_default(sender);
 
-                    let info = locked.application.get_room_info(room_id.to_owned());
-                    info.insert(ev.into_full_event(room_id.to_owned()));
+                    let ChatStore { rooms, previewer, .. } = &mut locked.application;
+                    let mut info = rooms.get_or_default(room_id.clone());
```

Comment:
@ulyssa `locked.application.get_room_info(room_id.to_owned())` and `rooms.get_or_default(room_id.clone())` should behave exactly the same, right? I only changed this to comply with the borrow checker.